source/features/filter-pr-by-build-status.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import {checkInline} from '../libs/icons';
+import features from '../libs/features';
+import {getOwnerAndRepo} from '../libs/utils';
+
+function init() {
+	const filterAssignee = select('.table-list-filters > .table-list-header-toggle:last-child > details:nth-child(7)');
+
+	filterAssignee.parentNode.insertBefore(
+		<details class="details-reset details-overlay float-left select-menu">
+			<summary class="btn-link select-menu-button" aria-haspopup="menu">
+				Status&nbsp;
+			</summary>
+			<details-menu class="select-menu-modal position-absolute right-0" style={{zIndex: 99}} role="menu">
+				<div class="select-menu-header">
+					<span class="select-menu-title">Filter by Status</span>
+				</div>
+				<div class="select-menu-list">
+					{getStatusDropDown()}
+				</div>
+			</details-menu>
+		</details>,
+		filterAssignee
+	);
+}
+
+const getStatusDropDown = () => {
+	const {ownerName, repoName} = getOwnerAndRepo();
+	const baseUrl = `https://github.com/${ownerName}/${repoName}/pulls`;
+
+	const statusList = ['Success', 'Failure', 'Pending'];
+
+	let existingFilter = select<HTMLInputElement>('#js-issues-search').value
+		.trim()
+		.replace(/\s+/, ' ');
+
+	const [currentStatus = ''] = existingFilter.match(/(?<=status:)(success|failure|pending)/) || [];
```

Comment:
```suggestion
	const [currentStatus] = existingFilter.match(/status:(success|failure|pending)/) || [undefined];
```

Reply:
Here, I'm getting the value of the status filter. So the change you have suggested will return `status:success` whereas my regex will return `success` only.

Reply:
Ah right, then you’d also need to change the first part to `const [, currentStatus]` because all groups are also captured separately

Reply:
Any specific reasons on why are we avoiding lookbehind? Compatibility? Performance?

Reply:
Also, For the suggestion you provided We'll have to provide two default values. What do you say?

```
const [, currentStatus] = queryString.match(/status:(success|failure|pending)/) || ['', ''];
```

Reply:
Regexes aren't very readable, so the shorter they are the better, especially if that means avoiding "advanced" features like lookbehind